node/src/components/rpc_server/http_server.rs

```diff
@@ -33,6 +62,24 @@ pub(super) async fn run<REv: ReactorEventT>(
     let rpc_get_auction_info = rpcs::state::GetAuctionInfo::create_filter(effect_builder);
     let rpc_get_rpcs = rpcs::docs::ListRpcs::create_filter(effect_builder);
 
+    // Catch requests where the method is not one we handle.
+    let unknown_method = warp::path(RPC_API_PATH)
+        .and(warp_json_rpc::filters::json_rpc())
+        .and_then(move |response_builder: warp_json_rpc::Builder| async move {
+            response_builder
+                .error(warp_json_rpc::Error::METHOD_NOT_FOUND)
+                .map_err(|_| warp::reject())
+        });
+
+    // Catch requests which don't parse as JSON.
+    let parse_failure = warp::path(RPC_API_PATH).and_then(move || async move {
+        let error_response = new_error_response(warp_json_rpc::Error::PARSE_ERROR);
+        Ok::<_, Rejection>(error_response)
+    });
```

Comment:
We won't know exactly _what_ error that was, right?

Reply:
Yes, that's correct.  The response contains no hints as to why the request couldn't be parsed.